Mail/mimePart.php

```diff
@@ -888,17 +893,19 @@ public static function encodeHeader($name, $value, $charset = 'ISO-8859-1',
                             if ($word[0] == '"' && $word[strlen($word)-1] == '"') {
                                 // de-quote quoted-string, encoding changes
                                 // string to atom
-                                $search = array("\\\"", "\\\\");
-                                $replace = array("\"", "\\");
-                                $word = str_replace($search, $replace, $word);
                                 $word = substr($word, 1, -1);
+                                $temp = preg_replace('/\\\\([\\\\"])/', '$1', $word);
+                                $word = ($temp !== NULL) ? $temp : $word;
```

Comment:
I'm not sure I like it. What's better, return an empty string on error or non-de-quoted? Anyway, preg_replace() error here would be really a rare case. I propose, to just use the return value and not check if it's null.